### PR TITLE
Add Beagle tool

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

This tool was requested by a user in GHelp: [original post](https://help.galaxyproject.org/t/phasing-snps-in-pedigrees-mini-linkage/6203?u=gallardoalba).

### CODE REVIEW

File: tools/beagle/beagle.xml [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        out=$out_prefix
+        nthreads=\${GALAXY_SLOTS:-1}
+        && gunzip 'out.vcf.gz'
+    ]]>    </command>
+    <inputs>
+        <param argument="gt" type="data" format="vcf" label="VCF file" 
+            help="It specifies a VCF file containing genotypes for the study samples. 
+                Each VCF record must contain a GT (genotype) format field"/>
+        <section name="optional_inputs" title="Optional input files" expanded="true">
+            <param argument="ref" type="data" format="vcf,binary" optional="true" label="Bref3 or VCF file with phased genotypes"
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/beagle/beagle.xml:61]: We can also just support VCF.

File: tools/beagle/beagle.xml [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+                VCF file when imputing ungenotyped markers. Genotype probabilities are calculated from allele 
+                probabilities assuming Hardy-Weinberg Equilibrium. Consequently, the alleles in the genotype 
+                with highest genotype probability may occasionally be different than the genotype obtained by 
+                taking the allele with highest probability on each haplotype, which is the genotype reported 
+                in the GT format field" />
+        </section>
+    </inputs>
+    <outputs>
+        <data name="vcf_file" format="vcf" from_work_dir="out.vcf" label="${tool.name} on ${on_string}: VCF file"/>
+        <data name="log_file" format="txt" from_work_dir="out.log" label="${tool.name} on ${on_string}: log file"/>
```
Comment (bgruening) [inline tools/beagle/beagle.xml:142]: make log file optional?

File: tools/beagle/beagle.xml [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    <expand macro='requirements' />
+    <command detect_errors='exit_code'><![CDATA[
+        #set out_prefix='out'
+        echo $optional_inputs.ref.ext &&
+        #if $optional_inputs.ref.ext == 'binary'
+            ln -s '${optional_inputs.ref}' ref.bref3 &&
+        #end if
+        beagle
+        gt='${gt}'
+        #if $optional_inputs.ref and $optional_inputs.ref.ext == 'binary'
```
Comment (bgruening) [inline tools/beagle/beagle.xml:16]: we do need to get rid of the binary here ... add a new filetype?

**Review by bgruening — APPROVED** [review]

File: tools/beagle/beagle.xml [hunk 4]
```diff
@@ -0,0 +1,315 @@
+<tool id='beagle' name='Beagle' version='@TOOL_VERSION@+galaxy@SUFFIX_VERSION@' profile='20.01'>
+    <description>phasing genotypes and imputing ungenotyped markers</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro='edam_ontology' />
+    <expand macro='requirements' />
+    <command detect_errors='exit_code'><![CDATA[
+        #set out_prefix='out'
+        echo $optional_inputs.ref.ext &&
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/beagle/beagle.xml:10]: Yes, just used it for testing.

File: tools/beagle/beagle.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/beagle/beagle.xml:61]: I can try to include a new filetype; the binary format is bref3 https://faculty.washington.edu/browning/beagle/bref3.14May18.pdf.

File: tools/beagle/beagle.xml (hunk 2 above)
Comment (gallardoalba) [reply tools/beagle/beagle.xml:142]: Sure.

File: tools/beagle/beagle.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/beagle/beagle.xml:61]: https://github.com/usegalaxy-eu/galaxy/pull/106

**Review by bgruening — APPROVED** [review]

**Pull request merged**